internal/mocks/mockgen_internal.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Mockgen refuses to generate mocks for internal packages.
+# This script copies the internal directory and renames it to internalpackage.
+# That way, mockgen can generate the mock.
+# Afterwards, it corrects the import paths (replaces internalpackage back to internal).
+
+TEMP_DIR=$(mktemp -d)
+mkdir -p $TEMP_DIR/src/github.com/lucas-clemente/quic-go/internalpackage
+
+cp -r $GOPATH/src/github.com/lucas-clemente/quic-go/internal/* $TEMP_DIR/src/github.com/lucas-clemente/quic-go/internalpackage
+find $TEMP_DIR -type f -name "*.go" -exec sed -i '' 's/internal/internalpackage/g' {} \;
```

Comment:
This doesn't work on linux, sed behaves non-standard on macOS iirc.

Reply:
I'm not planning to fix this. We're both working on OSX, so we can regenerate the mocks. This is the dirty work that mockgen is supposed to do.